### PR TITLE
Update Documentation

### PR DESCRIPTION
**Pull request by gingerbenw**
Separate documentation out into `docs` directory, and separate releasing guide into it's own document.

Updated for readability.

### CODE REVIEW

**Comment by gingerbenw** [comment]
Check the "rich diff" to see the flow chart for creating new branches

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -88,14 +88,34 @@ When a new Expo SDK is released, a new matching `@bugsnag/expo` version needs to
 
 ## Releases
 
-To start a release:
+### Enhancements and bugfixes
```
Comment (tomlongridge) [inline CONTRIBUTING.md]:
```suggestion
### Enhancements and bug fixes
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -88,14 +88,34 @@ When a new Expo SDK is released, a new matching `@bugsnag/expo` version needs to
...
 
-- create and push a new branch (e.g. `v46`) from the latest previous branch (e.g. `v45` or `v45-next`)
-- create a new next branch from which the release PR is to be made (e.g. `v46-next`)
-- make the required dependency and CLI changes (see above) for the latest expo version
-- regenerate the expo e2e test fixture using the expo create app cli
+- decide on a new version number, following [semantic versioning](https://semver.org/)
+- create a new release branch from `vX/next` with the new version number in the branch name i.e. `git checkout -b release/vX.Y.Z`
 - update the version number and date in the changelog
-- make a PR from your release branch (e.g. `v46-next`) to `v46` entitled `Release v46`
+- make a PR from your release branch to `vX/main` entitled Release vX.Y.Z
```
Comment (tomlongridge) [inline CONTRIBUTING.md]:
```suggestion
- make a PR from your release branch to `vX/main` entitled "Release vX.Y.Z"
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -88,14 +88,34 @@ When a new Expo SDK is released, a new matching `@bugsnag/expo` version needs to
...
+- get the release PR reviewed – all code changes should have been reviewed already, this should be a review of the integration of all changes to be shipped and the changelog
+- consider shipping a [prerelease](#prereleases) to aid testing the release
+
+### New major version
```
Comment (tomlongridge) [inline CONTRIBUTING.md]:
```suggestion
### New Expo SDK release
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -88,14 +88,34 @@ When a new Expo SDK is released, a new matching `@bugsnag/expo` version needs to
...
+graph TD;
+    v47/main-->v48/main;
+    v48/main-->v48/next;
+    v48/next-->PLAT-1234-support-v48;
+    PLAT-1234-support-v48-. PR .->v48/next;
+    v48/next-->release/v48;
+    release/v48-. PR .->v48/main;
+```
+
+- create and push a new **main** branch for the new supported expo version (e.g. `v48/main`) from the latest current branch (e.g. `v47/main`)
```
Comment (tomlongridge) [inline CONTRIBUTING.md]:
```suggestion
- create and push a new **main** branch for the new supported Expo version (e.g. `v48/main`) from the latest current branch (e.g. `v47/main`), checking for unreleased changes in the equivalent `next` branch (e.g. `v47/next`)
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -88,14 +88,34 @@ When a new Expo SDK is released, a new matching `@bugsnag/expo` version needs to
...
+- create a new **next** branch based on the new **main** branch (e.g. `v48/next`)
+- create a feature branch from which the changes to support the new version are to be made (e.g. `PLAT-1234-support-v48`)
+- make the required dependency and CLI changes for the latest expo version (see [Keeping dependencies in sync](#keeping-dependencies-in-sync))
```
Comment (tomlongridge) [inline CONTRIBUTING.md]: I think we should capitalise Expo as that's how they show it - please apply throughout the doc.

```suggestion
- make the required dependency and CLI changes for the latest Expo version (see [Keeping dependencies in sync](#keeping-dependencies-in-sync)) 
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -145,6 +165,10 @@ Finally:
     git push
     ```
 
+
```
Comment (tomlongridge) [inline CONTRIBUTING.md]: The final bullet needs re-writing for the proposed `vXX/next`/`main` format.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -145,6 +165,10 @@ Finally:
...
+
```
Comment (tomlongridge) [inline CONTRIBUTING.md:78]: Also we want to change the default branch on GitHub.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -88,14 +88,34 @@ When a new Expo SDK is released, a new matching `@bugsnag/expo` version needs to
...
-- make the required dependency and CLI changes (see above) for the latest expo version
-- regenerate the expo e2e test fixture using the expo create app cli
+- decide on a new version number, following [semantic versioning](https://semver.org/)
+- create a new release branch from `vX/next` with the new version number in the branch name i.e. `git checkout -b release/vX.Y.Z`
 - update the version number and date in the changelog
-- make a PR from your release branch (e.g. `v46-next`) to `v46` entitled `Release v46`
+- make a PR from your release branch to `vX/main` entitled Release vX.Y.Z
+- get the release PR reviewed – all code changes should have been reviewed already, this should be a review of the integration of all changes to be shipped and the changelog
+- consider shipping a [prerelease](#prereleases) to aid testing the release
+
```
Comment (tomlongridge) [inline CONTRIBUTING.md]: Should we also consider merging or cherry-picking the fix to other affected major version branches?

**Comment by imjoehaines** [comment]
We also need to update the @bugsnag/js pipeline so that it always triggers the latest 3 versions

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -1,13 +1,16 @@
...
 
 * [How to contribute](#how-to-contribute)
   * [Reporting issues](#reporting-issues)
   * [Fixing issues](#fixing-issues)
   * [Adding features](#adding-features)
 * [System requirements](#system-requirements)
+* [Keeping dependencies in sync](#keeping-dependencies-in-sync)
+* [New version of Expo](#updating-the-cli-to-install-a-compatible-notifier-version)
 * [Testing](#testing)
+* [Releases](#releases)
```
Comment (tomlongridge) [inline CONTRIBUTING.md]: I think this TOC is a bit out-of-date with the content

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -51,13 +53,13 @@ If you want to run the end-to-end tests locally you'll need [Docker](https://www
...
 
-## CI
+### CI
 
 CI runs on Buildkite. Tests are run automatically on any branch from within this repo. PRs from external repos do not run on the private test infrastructure. Once an external PR has been reviewed by a Bugsnag employee, a branch can be created within this repo in order to run on CI.
 
-⚠️ __Caution__: exercise due-diligence before creating a branch based on an external contribution – for example, be sure not to merge a bitcoin miner disguised as a bug fix!
+⚠️ __Caution__: exercise due-diligence before creating a branch based on an external contribution
 
 ## Keeping dependencies in sync
```
Comment (tomlongridge) [inline CONTRIBUTING.md]: How about making a section called "Supporting new Expo SDK versions" after "Adding new features" to contain all the bits around bumping the SDK?

We've also recently added an example, so we should consider what our approach for this should be – just keep the latest or have one per supported version, deleting the older ones. This should also be included in this section.

File: docs/RELEASING.md [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+
+When a new Expo SDK is released, we should also publish a matching major version.
+
+- create and push a new **main** branch for the new supported Expo version (e.g. `v48/main`) from the latest current branch (e.g. `v47/main`), checking for unreleased changes in the equivalent `next` branch (e.g. `v47/next`)
+- create a new **next** branch based on the new **main** branch (e.g. `v48/next`)
+- create a feature branch from which the changes to support the new version are to be made (e.g. `PLAT-1234-support-v48`)
+- make the required dependency and CLI changes for the latest Expo version (see [Keeping dependencies in sync](#keeping-dependencies-in-sync)) 
+- regenerate the e2e test fixture using the `create-expo-app` cli
+- update the version number and release date in the changelog
+- make a PR from your release branch (e.g. `PLAT-1234-support-v48` to `v48/main`) entitled `Release v48.0.0`
```
Comment (hamudi-ahmed) [inline docs/RELEASING.md]:
```suggestion
- regenerate the e2e test fixture using the `create-expo-app` cli
- make a PR from the feature branch targeting the new **next** branch (e.g. `PLAT-1234-support-v48` to `v48/next`)
- create a release branch from **next** (e.g. `release/v48.0.0`) 
- update the version number and release date in the changelog
- make a PR from your release branch targeting the new **main** branch (e.g. `release/v48.0.0` to `v48/main`) entitled `Release v48.0.0`
```

**Review by tomlongridge — APPROVED** [review]

**Pull request merged**